symforce/codegen/codegen_util.py

```diff
@@ -457,9 +462,7 @@ def _get_scalar_keys_recursive(
                 )
             )
     elif issubclass(datatype, sf.Matrix) or not use_data:
-        # TODO(hayk): Fix this in follow-up.
-        # TODO(nathan): I don't think this deals with 2D matrices correctly
-        if config.matrix_is_1d and config.use_eigen_types:
+        if config.use_eigen_types:
```

Comment:
For my own notes mostly, since it took me a bit to think through all this: the above changes don't affect matrices in LCM types, this change here affects only matrices in LCM types.  So the above versions assume the objects are the language-native matrix types, for which we want to use `format_matrix_accessor`.  Here we assume it's an LCM-type-internal matrix, which are represented by their storage.  This primarily assumes you don't try to call a symforce-generated function with an LCM type containing a language-native matrix type, which is possible to do e.g. in Python where you can put a `numpy.ndarray` into an LCM type where we're expecting an `eigen_lcm.Vector4d` or whatever.  This wasn't well supported before either though and seems fine to just not support at all, the one thing we might need to do is message that properly